app/src/main/java/org/mozilla/tv/firefox/pocket/PocketVideoRepo.kt

```diff
@@ -0,0 +1,129 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.pocket
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MutableLiveData
+import android.os.SystemClock
+import android.support.annotation.UiThread
+import kotlinx.coroutines.experimental.CancellationException
+import kotlinx.coroutines.experimental.Job
+import kotlinx.coroutines.experimental.delay
+import kotlinx.coroutines.experimental.launch
+import org.mozilla.tv.firefox.utils.BuildConfigDerivables
+import java.util.concurrent.TimeUnit
+import kotlin.math.min
+
+private val CACHE_UPDATE_FREQUENCY_MILLIS = TimeUnit.MINUTES.toMillis(45)
+
+/**
+ * Manages backing state for Pocket data, as well as any logic related to
+ * retrieving or storing that data.
+ */
+open class PocketVideoRepo(
+    private val pocketEndpoint: PocketEndpoint,
+    private val pocketFeedStateMachine: PocketFeedStateMachine,
+    buildConfigDerivables: BuildConfigDerivables
+) {
+
+    sealed class FeedState {
+        data class LoadComplete(val videos: List<PocketViewModel.FeedItem>) : FeedState()
+        object Loading : FeedState()
+        object NoKey : FeedState()
+        object FetchFailed : FeedState()
+    }
+
+    private val _feedState = MutableLiveData<FeedState>().apply {
+        // _feedState.value should always be initialized at the top of init,
+        // because we treat it as non-null
+        value = buildConfigDerivables.initialPocketRepoState
+    }
+    open val feedState: LiveData<FeedState> = _feedState
+
+    @Suppress("ANNOTATION_TARGETS_NON_EXISTENT_ACCESSOR") // Private properties generate fields so method annotations can't apply.
+    @get:UiThread
+    @set:UiThread
+    private var backgroundUpdates: Job? = null
+    @Volatile private var lastSuccessfulUpdateMillis = -1L
+    @Volatile private var lastUpdateAttemptMillis = -1L
+    @Volatile private var retryDelayMillis = 1_000L
+    @Volatile private var lastAttemptWasSuccessful = false
+
+    fun update() {
+        launch { updateInner() }
```

Comment:
Should a manual `update` reset the exponential backoff? Say a user clicks the "try again" button.

Reply:
Does this launch need an explicit scope with Kotlin 1.3?

Reply:
I think when we migrate it will make sense to make the repo itself the enclosing scope here, since we don't have any real case where we cancel this.  I'm not certain though, I need to read more about the changes.